File: fe-ta/src/models/ProductMedia.ts
export interface ProductMediaRequest {
  mediaFileUrl?: File | null;
  colorId?: number ;
  mediaType?: string; // "image" or "video"
  descriptionMedia?: string;
  isPrimary?: boolean;
  orderIndex?: number;
  note?: string;
}

export interface ProductMediaFilterRequest {
  productName?: string;
  colorName?: string;
  mediaFileUrl?: string;
  mediaType?: string; // "image" | "video"
  descriptionMedia?: string;
  isPrimary?: boolean;
  createdName?: string;
  updatedName?: string;
  fromUpdateTime?: string;
  toUpdateTime?: string;
  note?: string;
}

export interface ProductMediaResponse {
  id: number;
  productId: number;
  mediaFileUrl?: string;
  colorId: number ;
  colorName?: string;
  mediaType?: string;
  descriptionMedia?: string;
  isPrimary: boolean;
  orderIndex: number;
  createUid?: number;
  writeIUid?: number;
  createdName?: string;
  updatedName?: string;
  updateTime?: string;
  note?: string;
  option1?: string;
  option2?: string;
  option3?: string;
  option4?: string;
  option5?: string;
}

File: beSQLSugar/beSQLSugar/Enities/ProductMedia.cs
using SqlSugar;
namespace beSQLSugar.Models
{
    [SugarTable("product_media")]
    public class ProductMedia
    {
        [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
        public int Id { get; set; }

        [SugarColumn(IsNullable = false)]
        public int ProductId { get; set; }

        [SugarColumn(Length = 1250)]
        public string? MediaFileUrl { get; set; }

        [SugarColumn(Length = 20)]
        public string? MediaType { get; set; } // "image" or "video"

        [SugarColumn(DefaultValue = "0")]
        public bool IsPrimary { get; set; } = false;

        [SugarColumn(DefaultValue = "0")]
        public int OrderIndex { get; set; } = 0;

        [SugarColumn(IsNullable = true)]
        public int? CreateUid { get; set; }

        [SugarColumn(IsNullable = true)]
        public int? WriteIUid { get; set; }

        [SugarColumn(IsNullable = true)]
        public DateTime? UpdateTime { get; set; }

        [SugarColumn(Length = 250, IsNullable = true)]
        public string? Note { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option1 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option2 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option3 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option4 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option5 { get; set; }
    }
}
